### PR TITLE
Tickets/dm 1763

### PR DESCRIPTION
**Pull request by danielsf**

### CODE REVIEW

File: patches/tmvTable.patch [hunk 1]
```diff
@@ -0,0 +1,9 @@
+diff --git a/ups/tmv.table b/ups/tmv_LSST.table
+index 006ea09..b9f4d57 100644
+--- a/ups/tmv_LSST.table
++++ b/ups/tmv.table
+@@ -1,3 +1,4 @@
++setupRequired(scons)
+ envPrepend(PATH,${PRODUCT_DIR}/bin)
+ envPrepend(LD_LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
Comment (ktlim) [inline patches/tmvTable.patch:8]: Should have DYLD_LIBRARY_PATH as well here.

File: patches/tmvTable.patch [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+ envPrepend(LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
Comment (ktlim) [inline patches/tmvTable.patch]: What uses this variable?  It's not usually present.

File: ups/eupspkg.cfg.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+NJOBS=1
+export SCONSFLAGS=$SCONSFLAGS" USE_UNKNOWN_VARS=true PREFIX="$PREFIX"/lib"
```
Comment (ktlim) [inline ups/eupspkg.cfg.sh]: It's unusual to have a PREFIX point to a lib directory.  You should at least have a comment to explain this.

File: ups/tmv.table [hunk 4]
```diff
@@ -0,0 +1,7 @@
+setupRequired(scons)
+envPrepend(PATH,${PRODUCT_DIR}/bin)
+envPrepend(LD_LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
Comment (ktlim) [inline ups/tmv.table:3]: Add DYLD_LIBRARY_PATH here.

File: ups/tmv.table [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+envPrepend(LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
Comment (ktlim) [inline ups/tmv.table]: LIBRARY_PATH, C_INCLUDE_PATH, and CPATH are not typically used.  Are you sure they are necessary?

**Comment by danielsf** [comment]
Okay.  I wasn't sure what our policy was on changing the source code of external packages.

I agree that just removing the externally-provided table files will make life easier.

**Comment by ktlim** [comment]
Does TMV come with the ups directory as part of its distribution?  If so, then forget what I said (but it is unusual).

**Comment by ktlim** [comment]
Oh, I see, it's from Mike Jarvis.  That makes more sense.  I guess you may have to go back to what you were doing, if the packaging ups directory doesn't override the upstream ups directory.  Sorry.

**Comment by danielsf** [comment]
I am currently trying a build in which I deleted the native table file in the tarball.

The build seems to be working.  Assuming it works, is there any harm in providing a tarball from which we have extracted the native table file (I guess, if we ever upgrade to a future version, whoever does that will have to be warned that they will have to get rid of the native table file)?

**Comment by ktlim** [comment]
Yes, the problem is future upgrades.  We should leave the upstream distribution as is, to the extent possible.  You might recommend that your patches be applied by TMV so that we can remove them from our packaging.

**Comment by danielsf** [comment]
Okay

**Comment by danielsf** [comment]
On a similar note:

The LIBRARY_PATH, C_INCLUDE_PATH, and CPATH variables are all set by the native table file (which is why I included them in our table file).

I am testing a build in which I do not include them in our table file and I use the patches to remove them from the native table file.

If that passes, should I go with that, or err on the side of setting the environment variables?

**Comment by ktlim** [comment]
I would say that we should not unnecessarily set variables (although setting LD_LIBRARY_PATH on Darwin and DYLD_LIBRARY_PATH on non-Darwin violates that), so keep the patch.

**Pull request merged**